lib/net/ldap/entry.rb

```diff
@@ -196,4 +196,8 @@ def setter?(sym)
     sym.to_s[-1] == ?=
   end
   private :setter?
+
+  def ==(other)
+    return @myhash == other.to_h
```

Comment:
```suggestion
    @myhash == other.to_h
```

Also, should this method check if the `other` is the same class?

Reply:
While we don't really have an exposed way to create objects that inherit from Net::LDAP::Entry I would say those would be OK be OK to consider equal as far as Class goes. For [Ruby Hash objects](https://ruby-doc.org/core-2.7.2/Hash.html#method-i-3D-3D) `==` Returns true if all of the following are true:

> - object is a Hash object.
> - hash and object have the same keys (regardless of order).
> - For each key key, hash[key] == object[key].